services/src/test/scala/cromwell/services/metadata/impl/MetadataDatabaseAccessSpec.scala

```diff
@@ -473,9 +473,29 @@ class MetadataDatabaseAccessSpec extends FlatSpec with Matchers with ScalaFuture
       } yield()).futureValue
     }
 
+    it should "properly query metadata summaries based on archived status and timestamp and update archive status after metadata deletion" taggedAs DbmsTest in {
+      (for {
+        workflowId1 <- succeededWorkflowMetadata(WorkflowId.fromString("11111111-1111-1111-1111-111111111112"))
+        workflowId2 <- succeededWorkflowMetadata(WorkflowId.fromString("11111111-1111-1111-1111-111111111113"))
+        _ <- dataAccess.refreshWorkflowMetadataSummaries(1000, Option(Long.MaxValue))
+        _ <- dataAccess.updateMetadataArchiveStatus(workflowId1, MetadataArchiveStatus.Archived)
+        responseEmpty <- dataAccess.queryRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp(Option("Archived"), OffsetDateTime.now().minusMinutes(1))
+        _ = responseEmpty shouldBe empty
+        _ = Thread.sleep(60000)
+        responseNonEmpty <- dataAccess.queryRootWorkflowSummaryEntriesByArchiveStatusAndOlderThanTimestamp(Option("Archived"), OffsetDateTime.now().minusMinutes(1))
+        _ = responseNonEmpty should not be empty
+        _ = responseNonEmpty should contain allElementsOf Seq(workflowId1.toString)
+        _ = responseNonEmpty should contain noElementsOf Seq(workflowId2.toString)
+        _ <- dataAccess.deleteNonLabelMetadataEntriesForWorkflowAndUpdateArchiveStatus(workflowId1, Option("ArchivedAndPurged"))
+        summaryResponse <- dataAccess.queryWorkflowSummaries(WorkflowQueryParameters(Seq(WorkflowQueryKey.Id.name -> workflowId1.toString)))
+        _ = summaryResponse._1.results should not be empty
+        _ = summaryResponse._1.results.head.metadataArchiveStatus.toString shouldBe "ArchivedAndPurged"
+      } yield ()).futureValue(timeout = Timeout(3.minutes))
```

Comment:
TOL: I'm not a big fan of this `.futureValue` style of testing. I find it makes things harder than expected to debug when something fails in the middle.

How would you feel about making this test case read a little more line-by-line - eg like this?
```
val futureThingFuture = futureThing()
val futureThingValue = Await.result(futureThingFuture, 60.seconds)
futureThingValue should be(42)

val futureThingFuture2 = nextFuture(futureThingValue)
[...]
```